### ISSUE TITLE
Update to 1.12.7 bumps `minSdkVersion` to 26 on Android

### ISSUE THREAD

**Issue by AlvaroBrey** [mockk/mockk#893]
### Prerequisites

- [x] I am running the latest version
- [x] I checked the documentation and found no answer
- [x] I checked to make sure that this issue has not already been filed

### Expected Behavior

Upgrade to 1.12.7 from 1.12.5 in a project with `minSdk 23`, everything keeps working. I would expect no breaking changes due to this being a patch release.

### Current Behavior

Test build fails due to incompatible `minSdk`. See stacktrace.

#### Steps to Reproduce

1. Create an Android project with `minSdk = 23`
2. Use mockK 1.12.7

#### Context

* MockK version: 1.12.7
* OS: Linux
* Kotlin version: 1.6.21
* JDK version: 11.0.13
* JUnit version: 4.13.2
* Type of test: unit test OR android instrumented test: both

#### Stack trace

```
* What went wrong:
Execution failed for task ':app:processGenericDebugAndroidTestManifest'.
> Manifest merger failed : uses-sdk:minSdkVersion 23 cannot be smaller than version 26 declared in library [io.mockk:mockk-android:1.12.7] /home/user/.gradle/caches/transforms-3/0debc4558cb38f69fc40c4cac52d713c/transformed/jetified-mockk-android-1.12.7/AndroidManifest.xml as the library might be using APIs not available in 23
        Suggestion: use a compatible library with a minSdk of at most 23,
                or increase this project's minSdk version to at least 26,
                or use tools:overrideLibrary="io.mockk" to force usage (may lead to runtime failures)
```

**Comment by SimonMarquis** [comment]
The update was done here: https://github.com/mockk/mockk/commit/269a599fee17c43c634c5409e680e15ec40f00d6

**Comment by aSemy** [comment]
I've had a quick stab #901, can you check? 

If the GHA tests pass, can you try out the fix locally? Check out the branch from #901, deploy to Maven Local `./gradlew publishToMavenLocal`, bump your MockK versions, and give it a go

**Comment by AlvaroBrey** [comment]
> I've had a quick stab #901, can you check?
> 
> If the GHA tests pass, can you try out the fix locally? Check out the branch from #901, deploy to Maven Local `./gradlew publishToMavenLocal`, bump your MockK versions, and give it a go

I've verified that your PR fixes the situation for my setup :+1: . 
However keep in mind that before https://github.com/mockk/mockk/commit/269a599fee17c43c634c5409e680e15ec40f00d6, the `minSdk` was `21`, not `23`; so 1.12.7 will still be breaking for those on sdk `21` and `22`.

**Comment by aSemy** [comment]
@AlvaroBrey Thanks for noticing that it was 21, not 23!
